bundles/EcommerceFrameworkBundle/IndexService/ProductList/ElasticSearch/AbstractElasticSearch.php

```diff
@@ -1419,4 +1474,43 @@ public function valid()
 
         return $var;
     }
+
+    /**
+     * @return bool
+     */
+    public function isLoadFromSource(): bool
+    {
+        return $this->loadFromSource;
+    }
+
+    /**
+     * Use this method to set whether Elastic Search data should be loaded directly from source or not.
+     * @param bool $loadFromSource
+     */
+    public function setLoadFromSource(bool $loadFromSource)
+    {
+        $this->loadFromSource = $loadFromSource;
+    }
+
+    /**
+     * Get the score from a loaded product list based on a (Pimcore) product Id.
+     * Precondition: isLoadFromSource - mode must be set to true.
+     * @param int $productId the Pimcore product Id.
+     * @return float the score returned by Elastic Search.
+     * @throws \Exception if loadFromSource mode is not true.
+     */
+    public function getScoreFromLoadedList(int $productId): float
```

Comment:
what has that to do with the PR?

Reply:
Based on the new approach it is possible to access the score which can be very helpful for debugging search implementations. Until now the score was not accessible based on productLists.

If you prefer, we can either remove the method completely, or create another PR for it.
However, I am not 100% if this is the best place for the solution anyway, so probably we should discuss that matter in another issue or PR after this PR has been completed?

Reply:
let leave it there